### ISSUE TITLE
pyvolve.Model automatically updates a parameter dict

### ISSUE THREAD

**Issue by kfuku52** [sjspielman/pyvolve#21]
Today I found a strange behavior that might be unintentional. The pyvolve version is 1.0.2. This is a minimal code:
```
import pyvolve
import numpy as np

cmp = {'omega':0.11, 'kappa':2.089}
sf = np.ones((61))/61

model1 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)
print('model1:')
print(model1.matrix)
print('')
model2 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)
print('model2:')
print(model2.matrix)
print('')
```
The matrices of model 1 and model 2 should be identical because the input variables are the same. But the outputs do not match.
```
Using default codon model, GY-style.
model1:
[[-0.73109782  0.02595407  0.49289137 ...  0.          0.
   0.        ]
 [ 0.02595407 -0.75705189  0.02595407 ...  0.          0.
   0.        ]
 [ 0.49289137  0.02595407 -0.73109782 ...  0.          0.
   0.        ]
 ...
 [ 0.          0.          0.         ... -0.75705189  0.02595407
   0.49289137]
 [ 0.          0.          0.         ...  0.02595407 -1.16977114
   0.02595407]
 [ 0.          0.          0.         ...  0.49289137  0.02595407
  -0.75705189]]

Using default codon model, GY-style.
model2:
[[-0.8359948   0.01565538  0.62108026 ...  0.          0.
   0.        ]
 [ 0.01565538 -0.85165018  0.01565538 ...  0.          0.
   0.        ]
 [ 0.62108026  0.01565538 -0.8359948  ...  0.          0.
   0.        ]
 ...
 [ 0.          0.          0.         ... -0.85165018  0.01565538
   0.62108026]
 [ 0.          0.          0.         ...  0.01565538 -1.38875623
   0.01565538]
 [ 0.          0.          0.         ...  0.62108026  0.01565538
  -0.85165018]]
```

I found that the dict `cmp` was automatically updated after generating model1, and this was the cause of the discrepancy. When I redefine `cmp` as below, the identical matrices were returned.
```
cmp = {'omega':0.11, 'kappa':2.089}
sf = np.ones((61))/61

model1 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)
print('model1:')
print(model1.matrix)
print('')
cmp = {'omega':0.11, 'kappa':2.089}
model2 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)
print('model2:')
print(model2.matrix)
print('')
```

I'm not sure if it's intentional behavior, but I thought it should be reported.

**Comment by sjs-student** [comment]
Thanks, I will have a look at this shortly!

**Comment by sjspielman** [comment]
Sorry I had missed this issue during a busy few months back there. Dictionaries in python are fun like this - I will look into it shortly.

**Comment by sjspielman** [comment]
I've fixed this issue (deepcopy bug) in a new branch, `issues-19_21`, to be merged into a new release with a few bug fixes, shortly.

This code now gives the following output:

#### Script
```
import pyvolve
import numpy as np

cmp = {'omega':0.11, 'kappa':2.089}
sf = np.ones((61))/61

model1 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)
model2 = pyvolve.Model(model_type='codon', parameters=cmp, state_freqs=sf)

print(np.array_equal(model1.matrix, model2.matrix)) # should be True!
print(cmp) # should match original!
```

#### Output
```
True
{'omega': 0.11, 'kappa': 2.089}
```

**Comment by kfuku52** [comment]
Thank you!

**Comment by sjspielman** [comment]
Now fixed in new patch version `1.0.3`; you can update with `pip`. Thanks for catching this!!
